src/semeio/fmudesign/design_distributions.py

```diff
@@ -82,27 +82,29 @@ def _check_dist_params_uniform(dist_params: Sequence[str]) -> tuple[bool, str]:
     return status, msg
 
 
-def _check_dist_params_triang(dist_params: Sequence[str]) -> tuple[bool, str]:
+def validate_triangular_params(
+    dist_params: Sequence[int | str | float],
+) -> tuple[float, float, float]:
+    """Validate and convert triangular distribution parameters."""
     if len(dist_params) != 3:
-        status = False
-        msg = (
-            "Triangular distribution must have 3 parameters, "
-            "but had " + str(len(dist_params)) + " parameters. "
+        raise ValueError(
+            f"Triangular distribution requires exactly 3 parameters, got {len(dist_params)}"
         )
-    elif not all(is_number(param) for param in dist_params):
-        status = False
-        msg = "Parameters for triangular distribution must be numbers. "
-    elif not (
-        (float(dist_params[2]) >= float(dist_params[1]))
-        and (float(dist_params[1]) >= float(dist_params[0]))
-    ):
-        status = False
-        msg = "Triangular distribution must have: low <= mode <= high. "
-    else:
-        status = True
-        msg = ""
 
-    return status, msg
+    try:
+        low, mode, high = [float(p) for p in dist_params]
+    except (ValueError, TypeError) as e:
+        raise ValueError("All parameters must be convertible to numbers") from e
```

Comment:
maybe nice for the user to have the "got [{low}, {mode}, {high}]" present in all exceptions?

Reply:
I have added this.